enumerable_methods.rb

```diff
@@ -0,0 +1,217 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    ind = 0
+    for i in self
+      yield(i, ind)
+      ind += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << n if yield(n)
+    end
+    new_arr
+  end
+
+  def arg?(item, arg)
+    return item.is_a? arg if arg.is_a? Class
+    return !(item =~ arg).nil? if arg.is_a? Regexp
+
+    item == arg
+  end
+
+  def my_all?(arg = nil)
+    my_each do |item|
+      case block_given?
+      when false
+        (return false unless item) unless arg
+        (return false unless arg?(item, arg)) if arg
+      else
+        return false unless yield(item)
+      end
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    my_each do |item|
+      case block_given?
+      when false
+        return true if !arg && item
+        return true if arg && is_arg?(item, arg)
+      else
+        return true if yield(item)
+      end
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    my_each do |item|
+      case block_given?
+      when false
+        return false if !arg && item
+        return false if arg && is_arg?(item, arg)
+      else
+        return false if yield(item)
+      end
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+    if !arg.nil?
+      my_each do |n|
+        counter += 1 if n == arg
+      end
+    elsif block_given?
+      my_each do |n|
+        counter += 1 if yield(n)
+      end
+    else
+      counter = self.length
+    end
+    counter
+  end
+
+=begin
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << yield(n)
+    end
+    new_arr
+  end
+=end
+
+  # my_map modified to accept either block or proc (i.e., a named block)
+
+  def my_map(my_proc = nil)
+    return to_enum(:my_map) unless block_given?
+
+    new_arr = []
+    my_each_with_index do |n, ind|
+      new_arr[ind] = my_proc.nil? ? yield(n) : my_proc.call(n)
+    end
+    new_arr
+  end
+
+  def binary_operation(acc, num, bin_op)
+    case bin_op
+    when :+ then acc += num
+    when :- then acc -= num
+    when :* then acc *= num
+    when :/ then acc /= num
+    when :** then acc **= num
+    when :% then acc %= num
+    end
+    acc
+  end
+
+  def my_inject(acc = nil, symbol = nil)
+    # %i[+ - * / ** %].my_any?(acc)
+    # ignore any passed block if with both symbol and starting value
```

Comment:
Comments: delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))